Networking/Repository/AbstractRepository.swift

```diff
@@ -225,17 +238,25 @@ fileprivate extension AbstractRepository {
     func perform(
         method: NetworkingMethod,
         path: String,
-        parameters: [String: Any]?,
-        headers: [String: String]?) -> SignalProducer<RawDataResponse, RepositoryError> {
+        parameters: [String: Any]? = .none,
+        headers: [String: String]? = .none,
+        requiresSession: Bool) -> SignalProducer<RawDataResponse, RepositoryError> {
         guard let url = buildURL(path: path) else { return SignalProducer(error: .invalidURL) }
         
+        var headers = headers
+        if requiresSession {
+            guard _sessionManager.isLoggedIn else { return SignalProducer(error: .unauthenticatedSession) }
+            headers = headers ?? [:]
+            headers![authenticationHeaders.first!.key] = authenticationHeaders.first!.value
+        }
```

Comment:
Sorry I don't get this too much. Couldn't you just add the authenticationHeaders to the normal headers and use that? Something like `headers = (headers ?? [:]) + authenticationHeaders`.
Btw, I would add in the documentation somewhere (maybe in the headers parameter in each function or in the AbstractRepository class description) that the authentication headers don't need to be passed, the abstract repository will handle them automatically with the session service, that "headers" is just for extra non-authentication headers.